backend/projects/views.py

```diff
@@ -4225,3 +4225,74 @@ def change_task_language_field_to_project_target_language(self, request, pk):
             {"message": "language field of task data succesfully updated!"},
             status=status.HTTP_200_OK,
         )
+
+    @action(
+        detail=True,
+        methods=["POST"],
+        url_name="set_password",
+    )
+    def set_password(self, request, pk=None):
+        try:
+            project = Project.objects.get(pk=pk)
+            password = request.data.get("password")
```

Comment:
There can be a case where password is absent in request.data, please handle that